web/js/addContact.js

```diff
@@ -1,3 +1,277 @@
+            column1.append(operatorCodeInput);
+            column1.append(countryCodeInput);
+            column1.append(phoneNumberInput);
+            column1.append(phoneTypeInput);
+            column1.append(commentaryInput);
+            column2.innerHTML = countryCode.concat(operatorCode, phoneNumber);
+            column3.innerHTML = phoneType;
+            column4.innerHTML = commentary;
+            column5.append(button);
+        } else {
+            alert("Phone exists");
+        }
+        document.getElementById('countryCodeInput').value = "";
+        document.getElementById('operatorCodeInput').value = "";
+        document.getElementById('phoneNumberInput').value = "";
+        document.getElementById('phoneTypeInput').value = "";
+        document.getElementById('commentaryInput').value = "";
+    }
+}
+
+function checkPhoneForUnique(phone) {
+    let allNumber = document.getElementsByName('phoneNumber');
+    for (let i = 0; i < allNumber.length; i++) {
+        if (allNumber[i].value === phone)
+            return false;
+    }
+    return true;
+}
+
+function deleteNumbers() {
+    let amountOfDeleted = 0;
+    let checkBoxes = document.getElementsByName('phoneIdForDelete');
+    let length = checkBoxes.length;
+    for (let i = 0; i < length; i++) {
+        let box = checkBoxes[i - amountOfDeleted];
+        if (box.checked === true) {
+            if (box.value === "new") {
+                document.getElementById('phoneTable').deleteRow(i + 1 - amountOfDeleted);
+                amountOfDeleted++;
+            } else {
+                let table = document.getElementById('phoneTable');
+                let row = table.rows[i - amountOfDeleted + 1];
+                document.getElementById('undoButton').style.visibility = "visible";
+                for (let j = 1; j < row.cells.length; j++) {
+                    row.cells[j].style.setProperty("text-decoration", "line-through");
+                }
+                if (!phonesForDelete.includes(box.value))
+                    phonesForDelete.push(box.value);
+                box.checked = false;
+            }
+        }
+    }
+}
+
+function undoDelete() {
+    let checkBoxes = document.getElementsByName('phoneIdForDelete');
+
+    for (let i = 0; i < checkBoxes.length; i++) {
+        let box = checkBoxes[i];
+        if (box.checked === true) {
+            let id = box.value;
+            const index = phonesForDelete.indexOf(id);
+            if (index !== -1) {
+                phonesForDelete.splice(index, 1);
+                unCrossTableRow(i + 1);
+            }
+        }
+        box.checked = false;
+    }
+}
+
+function unCrossTableRow(rowIndex) {
+    let table = document.getElementById('phoneTable');
+    let row = table.rows[rowIndex];
+    for (let j = 1; j < row.cells.length; j++) {
+        row.cells[j].style.setProperty("text-decoration", "none");
+    }
+}
+
+function submitAll() {
+    if (!isFieldEmpty(document.getElementById('name').value)) {
```

Comment:
I guess you can come up with more generic validation logic. For example create object where key - it's a field id and value is an array which contents validation functions which should be invoked one by one. 
```
{
    surname: [isFieldEmpty],
    email: [isFieldsEmpty, isValidEmail]
    .....
}
```
And `isFieldEmpty` function  may return error message or false if it's valid.
And we agreed with you that I will see this message under input field, not in alert window.